internal/newusers/manager.go

```diff
@@ -0,0 +1,338 @@
+	for _, g := range u.Groups {
+		// System group: ignore here, not part of the cache.
+		if g.GID == nil {
+			continue
+		}
+		groupContents = append(groupContents, cache.GroupDB{
+			Name: g.Name,
+			GID:  *g.GID,
+		})
+	}
+
+	// Update user information in the cache.
+	userDB := cache.UserDB{
+		UserPasswdShadow: cache.UserPasswdShadow{
+			Name:           u.Name,
+			UID:            u.UID,
+			GID:            *u.Groups[0].GID,
+			Gecos:          u.Gecos,
+			Dir:            u.Dir,
+			Shell:          u.Shell,
+			LastPwdChange:  -1,
+			MaxPwdAge:      -1,
+			PwdWarnPeriod:  -1,
+			PwdInactivity:  -1,
+			MinPwdAge:      -1,
+			ExpirationDate: -1,
+		},
+		LastLogin: time.Now(),
+	}
+	if err := m.cache.UpdateUserEntry(userDB, groupContents); err != nil {
+		return m.evaluateError(err)
+	}
+
+	return nil
+}
+
+// BrokerForUser returns the broker ID for the given user.
+func (m *Manager) BrokerForUser(username string) (string, error) {
+	brokerID, err := m.cache.BrokerForUser(username)
+	if err != nil {
+		return "", m.evaluateError(err)
+	}
+	return brokerID, err
+}
+
+// UpdateBrokerForUser updates the broker ID for the given user.
+func (m *Manager) UpdateBrokerForUser(username, brokerID string) error {
+	return m.evaluateError(m.cache.UpdateBrokerForUser(username, brokerID))
+}
+
+// UserByName returns the user information for the given user name.
+func (m *Manager) UserByName(username string) (cache.UserPasswdShadow, error) {
+	usr, err := m.cache.UserByName(username)
+	if err != nil {
+		return cache.UserPasswdShadow{}, m.evaluateError(err)
+	}
+	return usr, nil
+}
+
+// UserByID returns the user information for the given user ID.
+func (m *Manager) UserByID(uid int) (cache.UserPasswdShadow, error) {
+	usr, err := m.cache.UserByID(uid)
+	if err != nil {
+		return cache.UserPasswdShadow{}, m.evaluateError(err)
+	}
+	return usr, nil
+}
+
+// AllUsers returns all users.
+func (m *Manager) AllUsers() ([]cache.UserPasswdShadow, error) {
+	usrs, err := m.cache.AllUsers()
+	if err != nil {
+		return nil, m.evaluateError(err)
+	}
+	return usrs, err
+}
+
+// GroupByName returns the group information for the given group name.
+func (m *Manager) GroupByName(groupname string) (cache.Group, error) {
+	grp, err := m.cache.GroupByName(groupname)
+	if err != nil {
+		return cache.Group{}, m.evaluateError(err)
+	}
+	return grp, nil
+}
+
+// GroupByID returns the group information for the given group ID.
+func (m *Manager) GroupByID(gid int) (cache.Group, error) {
+	grp, err := m.cache.GroupByID(gid)
+	if err != nil {
+		return cache.Group{}, m.evaluateError(err)
+	}
+	return grp, nil
+}
+
+// AllGroups returns all groups.
+func (m *Manager) AllGroups() ([]cache.Group, error) {
+	grps, err := m.cache.AllGroups()
+	if err != nil {
+		return nil, m.evaluateError(err)
+	}
+	return grps, nil
+}
+
+// evaluateError checks the error and requests a database clearing if needed.
+func (m *Manager) evaluateError(err error) error {
```

Comment:
I would be more direct in the name `shouldClearDatabase` or something else the long?